### PR TITLE
Fix #15903: Added MusicXML support for some accidentals

### LINKED ISSUE #15903: [MU4 Issue] Extended Helmholtz-Ellis SMuFL accidentals not imported correctly from MusicXML 4 files
**Describe the bug**
MuseScore 4 does not import Extended Helmholtz-Ellis SMuFL accidentals from musicXML 4 files.

**To Reproduce**
Steps to reproduce the behaviour:
Open the attached musicXML file in MuseScore.

**Expected behaviour**
Pitches 2-12 in the sequence should contain the SMuFL accidentals defined in the attached musicXML file. The accidentals are (respectively):

accidentalFlatThreeArrowsUp
accidentalFlatOneArrowUp
accidentalFlatTwoArrowsUp
accidentalNaturalTwoArrowsUp
accidentalSharpTwoArrowsUp
accidentalSharpTwoArrowsUp
accidentalNaturalThreeArrowsDown
accidentalSharpTwoArrowsDown
accidentalNaturalTwoArrowsDown
accidentalFlatThreeArrowsDown
accidentalNaturalThreeArrowsUp

**Platform information**
OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.0.0-223472200, revision: github-musescore-musescore-5485621

[musicXML_file.zip](https://github.com/musescore/MuseScore/files/10421256/musicXML_file.zip)
[MuseScore_file.zip](https://github.com/musescore/MuseScore/files/10421257/MuseScore_file.zip)

### PR DESCRIPTION
**Pull request by HemantAntony**
Resolves: #15903 

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Comment by HemantAntony** [comment]
@DmitryArefiev can you test this? :) (Issue #15903)

File: src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp [hunk 1]
```diff
@@ -33,6 +33,104 @@
...
+    { "accidentalSharpEqualTempered",                    AccidentalType::SHARP_EQUAL_TEMPERED },
+    { "accidentalDoubleSharpEqualTempered",              AccidentalType::DOUBLE_SHARP_EQUAL_TEMPERED },
+    { "accidentalQuarterFlatEqualTempered",              AccidentalType::QUARTER_FLAT_EQUAL_TEMPERED },
+    { "accidentalQuarterSharpEqualTempered",             AccidentalType::QUARTER_SHARP_EQUAL_TEMPERED }
+};
+
+const static QMap<QString, SymId> smuflAccidentalTypesyms {
+    { "accidentalDoubleFlatOneArrowDown",               SymId::accidentalDoubleFlatOneArrowDown },
+    { "accidentalFlatOneArrowDown",                     SymId::accidentalFlatOneArrowDown },
+    { "accidentalNaturalOneArrowDown",                  SymId::accidentalNaturalOneArrowDown },
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp:89]: Wouldn't it be possible to use `SymNames::symIdByName(…)` for this?

Comment (HemantAntony) [reply src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp:89]: That should work. I didn't know a method existed for that 😁

**Comment by DmitryArefiev** [comment]
Tested #15903 on Win10 - **looks fine**

File: src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp [hunk 2]
```diff
@@ -506,18 +557,27 @@ QString accSymId2MxmlString(const SymId id)
     case SymId::accidentalKoron:                 s = "koron";
         break;
     default:
-        //s = "other"; // actually pick up the SMuFL name or SymId
+        s = "other";
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp:560]: What about the `"sharp1"` to `"flat4"` abowe?

File: src/importexport/musicxml/internal/musicxml/musicxmlsupport.h [hunk 3]
```diff
@@ -236,9 +236,11 @@ extern void domError(const QDomElement&);
 extern void domNotImplemented(const QDomElement&);
 
 extern QString accSymId2MxmlString(const SymId id);
+extern QString accSymId2SmuflMxmlString(const SymId id);
 extern QString accidentalType2MxmlString(const AccidentalType type);
-extern AccidentalType mxmlString2accidentalType(const QString mxmlName);
-extern SymId mxmlString2accSymId(const QString mxmlName);
+extern QString accidentalType2SmuflMxmlString(const AccidentalType type);
+extern AccidentalType mxmlString2accidentalType(const QString mxmlName, const QString smufl);
+extern SymId mxmlString2accSymId(const QString mxmlName, const QString smufl = ""); //#
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/musicxmlsupport.h:243]: What's that comment for?

File: src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp:560]: Should be
```cpp
            case SymId::accidental1CommaSharp:           s = "sharp1";               break;
            case SymId::accidental2CommaSharp:           s = "sharp2";               break;
            case SymId::accidental3CommaSharp:           s = "sharp3";               break;
            case SymId::accidental5CommaSharp:           s = "sharp5";               break;
            case SymId::accidental1CommaFlat:            s = "flat1";                break;
            case SymId::accidental2CommaFlat:            s = "flat2";                break;
            case SymId::accidental3CommaFlat:            s = "flat3";                break;
            case SymId::accidental4CommaFlat:            s = "flat4";                break;
```

File: src/importexport/musicxml/internal/musicxml/musicxmlsupport.h (hunk 3 above)
Comment (HemantAntony) [reply src/importexport/musicxml/internal/musicxml/musicxmlsupport.h:243]: oh That was mark for me. I forgot to remove it :)

**Comment by Jojo-Schmitz** [comment]
I'd have a backport to 3.x almost ready, but am struggling with the musicxml export part, help on that would be much appreciated

File: src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/musicxmlsupport.cpp:560]: Ping...

File: src/importexport/musicxml/internal/musicxml/musicxmlsupport.h (hunk 3 above)
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/musicxmlsupport.h:243]: Ping ;-)

**Review by RomanPudashkin — APPROVED** [review]

**Comment by HemantAntony** [comment]
@Jojo-Schmitz I updated the PR. Sure, what do you need help with? :)
@RomanPudashkin

**Comment by Jojo-Schmitz** [comment]
Thanks.
I'd need help with the import part of this PR and its backport to 3.x

**Comment by Jojo-Schmitz** [comment]
I meant export

**Comment by lvinken** [comment]
Hi Jojo, do you still need help with the backport to 3.x and if so, what exactly ?

**Comment by Jojo-Schmitz** [comment]
Yes, and with the export part of it, in https://github.com/Jojo-Schmitz/MuseScore/pull/131

**Pull request merged**